File: Standard/C/Cpp/1.cpp
/**
 * Create Time: 2020.11.25
 * By: LovelyAnQi
 */
#include <iostream>
using namespace std;

#define ull unsigned long long

ull currentPrime = 1;

ull nextPrime() {

	for (++currentPrime;; ++currentPrime) {
		ull i;
		for (i = 2; i * i < currentPrime; ++i) {
			if (currentPrime % i == 0) {
				break;
			}
		}
		if (i * i > currentPrime) {
			break;
		}
	}
	return currentPrime;
}

int main()
{
	ull x, mul = 1, k = 0;
	cin >> x;

	while (mul < x) {
		mul *= nextPrime();
		++k;
	}

	cout << k << endl;

	return 0;
}

File: Standard/C/Python3/1.py
# Create Time: 2020.11.25
# By: LovelyAnQi

currentPrime = 1

def nextPrime():
    global currentPrime
    currentPrime += 1
    while True:
        i = 2
        while i * i < currentPrime:
            if currentPrime % i == 0:
                break
            i += 1
        if i * i > currentPrime:
            break
        currentPrime += 1
    return currentPrime

x = int(input())
mul = 1
k = 0

while mul < x:
    mul *= nextPrime()
    k += 1

print(k)
